src/codegen/abi.js

```diff
@@ -242,27 +247,40 @@ module.exports = class AbiCodeGenerator {
                         member
                           .get('outputs')
                           .get(0)
-                          .get('type')
+                          .get('type'),
                       )
                     : `new ${returnType.name}(
                   ${member
                     .get('outputs')
                     .map((output, index) =>
                       typesCodegen.ethereumValueToAsc(
                         `result[${index}]`,
-                        output.get('type')
-                      )
+                        output.get('type'),
+                      ),
                     )
                     .join(', ')}
                 )`
                 };
-                `
-              )
+                `,
+              ),
             )
           }
       }
     })
 
     return [...types, klass]
   }
+
+  _indexedInputType(inputType) {
+    // strings, bytes and arrays are encoded and hashed to a bytes32 value
+    if (
+      inputType === 'string' ||
+      inputType === 'bytes' ||
+      inputType.match(/\[[0-9]*\]$/g)
```

Comment:
I assume that this is matching arrays, but from the regex it's not obvious to me.

Reply:
You assume correctly! I'll add a comment before I merge.